scipy/stats/stats.py

```diff
@@ -1080,7 +1082,8 @@ def _moment(a, moment, axis):
             n_list.append(current_n)
 
         # Starting point for exponentiation by squares
-        a_zero_mean = a - np.mean(a, axis, keepdims=True)
+        mean = a.mean(axis, keepdims=True) if mean is None else mean
```

Comment:
Move line 1085 close to the beginning of this function around line 1049. The mean could be None and it should be handled earlier.

Reply:
The mean is only used in this branch, so moving this line to the start of the function would drop performance of 1st and 0th moments.